derive/impl/src/lib.rs

```diff
@@ -197,15 +197,15 @@ pub fn derive_shader_type(input: DeriveInput, root: &Path) -> TokenStream {
                 align: None,
             };
             for attr in &field.attrs {
-                let span = attr.tokens.span();
-                if attr.path.is_ident("align") {
+                let span = attr.to_token_stream().span();
+                if attr.path().is_ident("align") {
```

Comment:
I don't think this is correct and is why we seem to lose the span info.

We should use the new `attr.meta.require_list()` and work with `MetaList`s `.path` and `.tokens.span()`.